File: Backend/Adapters/Services/AIAgents.Laboratory.API.Adapters/Models/Response/AgentsWorkspaceDTO.cs
using AIAgents.Laboratory.API.Adapters.Models.Base;
using AIAgents.Laboratory.API.Adapters.Models.Request;

namespace AIAgents.Laboratory.API.Adapters.Models.Response;

/// <summary>
/// The Agents Workspace DTO model.
/// </summary>
/// <seealso cref="BaseModelDTO"/>
public sealed record AgentsWorkspaceDTO : BaseModelDTO
{
    /// <summary>
    /// Gets or sets the agent workspace guid id.
    /// </summary>
    public string AgentWorkspaceGuid { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the agent workspace name.
    /// </summary>
    public string AgentWorkspaceName { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the active agents name and active agents guids.
    /// </summary>
    public IEnumerable<WorkspaceAgentsDataDTO> ActiveAgentsListInWorkspace { get; set; } = [];

    /// <summary>
    /// Gets or sets the active users in the workspace.
    /// </summary>
    public IEnumerable<string> WorkspaceUsers { get; set; } = [];

    /// <summary>
    /// Gets or sets a value indicating whether group chat is enabled.
    /// </summary>
    /// <value>The value indicating whether group chat is enabled.</value>
    public bool IsGroupChatEnabled { get; set; }
}

File: UI/web/models/response/agents-workspace-dto.ts
import { WorkspaceAgentsDataDTO } from "./workspace-agents-data.dto";

export class AgentsWorkspaceDTO {
	agentWorkspaceGuid: string = "";
	agentWorkspaceName: string = "";
	activeAgentsListInWorkspace: WorkspaceAgentsDataDTO[] = [];
	workspaceUsers: string[] = [];
	isGroupChatEnabled: boolean = false;
	dateCreated: Date = new Date();
	createdBy: string = "";
	dateModified: Date = new Date();
	modifiedBy: string = "";
}
